zerver/views/registration.py

```diff
@@ -691,7 +691,10 @@ def registration_helper(
             )
         assert realm is not None
 
-        full_name = form.cleaned_data["full_name"]
+        if realm_creation and create_demo:
+            full_name = "Your name"
```

Comment:
I think we may want generated default names to be something more like "Curious Crocodile" long term; but that can be discussed as a follow-up change.